x-pack/plugins/infra/public/pages/logs/settings/source_configuration_form_errors.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut, EuiCode } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React from 'react';
+import { FormValidationError } from './validation_errors';
+
+export const LogSourceConfigurationFormErrors: React.FC<{ errors: FormValidationError[] }> = ({
+  errors,
+}) => (
+  <EuiCallOut color="danger" iconType="alert" title={logSourceConfigurationFormErrorsCalloutTitle}>
+    <ul>
+      {errors.map((error, errorIndex) => (
+        <li key={errorIndex}>
+          <LogSourceConfigurationFormError error={error} />
+        </li>
+      ))}
+    </ul>
+  </EuiCallOut>
+);
+
+export const LogSourceConfigurationFormError: React.FC<{ error: FormValidationError }> = ({
+  error,
+}) => {
+  if (error.type === 'generic') {
```

Comment:
Is there a reason to have the human readable messages put together here, as opposed to the `FormValidationError` holding the text directly?

Reply:
Makes sense, thanks 👌